### PR TITLE
review needed for generate_fixtures.py only

### PR DESCRIPTION
**Pull request by abdelrahman725**
## Summary

…

## References

…

## Reviewer guidance

…

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/content/management/commands/generate_fixtures.py [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+
+        app_name = options["app"]
+        file_name = options["file_name"]
+        generating_mode = options["mode"]
+        required_levels = options["levels"]
+
+        apps_names = ("content", "lessons", "exams", "logger")
+        modes_option = ("build_fixtures", "default_db")
+
+        if generating_mode not in modes_option:
```
Comment (jredrejo) [inline kolibri/core/content/management/commands/generate_fixtures.py]: You don't need to add this code if you use the right argparse type https://docs.python.org/2.7/library/argparse.html#choices

File: kolibri/core/content/management/commands/generate_fixtures.py [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+        MIN_SCHEMA_VERSION = 1
+        DEVELOPER_NAME = "bedo khaled"
+
+        # for returning random choices
+        def choices(sequence, k):
+            return [random.choice(sequence) for _ in range(0, k)]
+
+        # puprpose: if we have a content node of certain kind what type of file (file_preset) should maps to this node  ?
+        content_kind_to_file_preset = {}
+        for format_object in format_presets.PRESETLIST:
```
Comment (jredrejo) [inline kolibri/core/content/management/commands/generate_fixtures.py]: you can simplify this code creating previusly a list:
```
allowed_formats = set(format_presets.PRESETLIST).difference(IGNORED_KINDS))
for format_object in allowed_formats:
      if format_object.kind not in content_kind_to_file_preset:
          content_kind_to_file_preset[format_object.kind] = [format_object.id]
      else:
          content_kind_to_file_preset[format_object.kind].append(
              format_object.id
          )
```

File: kolibri/core/content/migrations/0031_automatic_max_length.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
```
Comment (jredrejo) [inline kolibri/core/content/migrations/0031_automatic_max_length.py]: I don't understand why this migration is needed

File: kolibri/core/content/migrations/0032_invalid.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
```
Comment (jredrejo) [inline kolibri/core/content/migrations/0032_invalid.py]: same for this migration, what model changes are producing this migration?

**Review by jredrejo — CHANGES_REQUESTED** [review]
There are some required changes, specially for the migrations.
For the code there are room to improve it a bit, I left a couple of suggestions.
Also, currently it does not support logger, exams, etc. so I'd leave a comment when this option is defined.

**Comment by abdelrahman725** [comment]
> There are some required changes, specially for the migrations. For the code there are room to improve it a bit, I left a couple of suggestions. Also, currently it does not support logger, exams, etc. so I'd leave a comment when this option is defined.

yes, it's currently for the content app only, but it has the potenial to be for any app as we can notice by the dynamic way of getting anny app name

File: kolibri/core/content/management/commands/generate_fixtures.py (hunk 1 above)
Comment (abdelrahman725) [reply kolibri/core/content/management/commands/generate_fixtures.py]: done !

File: kolibri/core/content/management/commands/generate_fixtures.py (hunk 2 above)
Comment (abdelrahman725) [reply kolibri/core/content/management/commands/generate_fixtures.py]: i didn't quiet understand your code purpose, and  the first line throw error 'unhashable type list'

**Pull request closed without merging**